File: resharper/test/data/References/CodeCompletion/ListsMemberDataCandidatesInSameClass.cs
using System.Collections.Generic;
using Xunit;

namespace Foo
{
    public class MyTest
    {
        [Theory]
        [MemberData("{caret}")]
        public void Test1(int value)
        {
            Assert.Equal(42, value);
        }

        public static IEnumerable<object[]> DataEnumerator
        {
            get { yield return new object[] { 42 }; }
        }

        public static IEnumerable<object[]> AnotherDataEnumerator
        {
            get { yield return new object[] { 42 }; }
        }

        public static IEnumerable<object[]> MoreDataEnumerator
        {
            get { yield return new object[] { 42 }; }
        }

        public static IEnumerable<object[]> GetDataMethod()
        {
            yield return new object[] { 42 };
        }

        public static IEnumerable<object[]> GetDataField = new List<object[]> {
            new object[] { 42 }
        };
    }
}

File: resharper/test/data/References/CodeCompletion/ListsMemberDataCandidatesInSameClass.vb
Imports System
Imports System.Collections.Generic
Imports Xunit

Public Class MyTest
    <Theory>
    <MemberData("{caret}")>
    Public Sub Test1(ByVal value As Int32)
        Assert.Equal(42, value)
    End Sub

    Public Shared ReadOnly Iterator Property DataEnumerator() As IEnumerable(Of Object())
        Get
            Yield New Object() { 42 }
        End Get
    End Property

    Public Shared ReadOnly Iterator Property AnotherDataEnumerator() As IEnumerable(Of Object())
        Get
            Yield New Object() { 42 }
        End Get
    End Property

    Public Shared ReadOnly Iterator Property MoreDataEnumerator() As IEnumerable(Of Object())
        Get
            Yield New Object() { 42 }
        End Get
    End Property

    Public Shared Iterator Function GetDataMethod() As IEnumerable(Of Object())
        Yield New Object() { 42 }
    End Function

    Public Shared GetDataField As IEnumerable(Of Object())
End Class
